src/test/java/cz/fi/muni/pa165/esports/dao/PlayerDaoTest.java

```diff
@@ -0,0 +1,199 @@
+
+import cz.fi.muni.pa165.esports.PersistenceSampleApplicationContext;
+import cz.fi.muni.pa165.esports.entity.Player;
+import cz.fi.muni.pa165.esports.enums.Gender;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
+import org.springframework.transaction.annotation.Transactional;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import javax.persistence.*;
+import java.util.List;
+
+
+/**
+ * Tests for {@link PlayerDao}
+ * @author Gabriela Kandova
+ */
+
+@Transactional
+@ContextConfiguration(classes = {PersistenceSampleApplicationContext.class})
+public class PlayerDaoTest extends AbstractTestNGSpringContextTests {
+    @PersistenceUnit
+    protected EntityManagerFactory emf;
+
+    @PersistenceContext
+    protected EntityManager em;
+
+    @Autowired
+    private PlayerDao playerDao;
+
+    private Player mrWhite;  // ok
+    private Player mrOrange; // ok
+    private Player mrPink; // no name
+    private Player mrBrown; // not persisted
+    private Player mrBlonde; // ok
+    private Player mrsRed; // ok; mrWhite's namesake
+
+    @BeforeClass
+    private void setUp() {
+        em = emf.createEntityManager();
+        em.getTransaction().begin();
+
+        mrWhite = new Player();
+        mrWhite.setName("Larry");
+        mrWhite.setAge(50);
+        mrWhite.setGender(Gender.MALE);
+
+        mrOrange = new Player();
+        mrOrange.setName("Freddy");
+        mrOrange.setAge(30);
+        mrOrange.setGender(Gender.MALE);
+        em.persist(mrOrange);
+
+        mrPink = new Player();
+        mrPink.setName(null);
+        mrPink.setAge(30);
+        mrPink.setGender(Gender.MALE);
+
+        mrBrown = new Player();
+        mrBrown.setName("Mr. Brown");
+        mrBrown.setAge(28);
+        mrBrown.setGender(Gender.MALE);
+
+        mrBlonde = new Player();
+        mrBlonde.setName("Mr. Blonde");
+        mrBlonde.setAge(32);
+        mrBlonde.setGender(Gender.MALE);
+        em.persist(mrBlonde);
+
+        mrsRed = new Player();
+        mrsRed.setName(mrWhite.getName());
+        mrsRed.setAge(40);
+        mrsRed.setGender(Gender.FEMALE);
+        em.persist(mrsRed);
+
+        em.getTransaction().commit();
+    }
+
+    @Test
+    public void testCreate() {
+        playerDao.create(mrWhite);
+
+        Player foundPlayer = em.find(Player.class, mrWhite.getPlayer_id());
+
+        Assert.assertNotNull(foundPlayer);
+        Assert.assertEquals(mrWhite, foundPlayer);
+        Assert.assertEquals(foundPlayer.getName(), mrWhite.getName());
+        Assert.assertEquals(foundPlayer.getAge(), mrWhite.getAge());
+        Assert.assertEquals(foundPlayer.getGender(), mrWhite.getGender());
+    }
+
+    @Test(expectedExceptions = { PersistenceException.class })
```

Comment:
I'm not sure if this is the right exception, but it's the one that gets thrown so... (It also depends on the configuration, there is a bean that translates exceptions)